cpu/rpx0xx/periph/timer.c

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (C) 2021 Otto-von-Guericke Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_rp2040
+ * @ingroup     drivers_periph_timer
+ * @{
+ *
+ * @file
+ * @brief       Timer implementation for the RPX0XX
+ * @details     The RPX0XX has a 64 bit µs timer but timer interrupts match
+ *              on the lower 32 bits.
+ *
+ * @author      Fabian Hüßler <fabian.huessler@ovgu.de>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <assert.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "vendor/RP2040.h"
+#include "io_reg.h"
+#include "timex.h"
+#include "periph_conf.h"
+#include "periph/timer.h"
+
+#define DEV(d)          (timer_config[d].dev)
+#define ALARM(d, a)     ((&(DEV(d)->ALARM0)) + (a))
+
+static timer_cb_t _timer_ctx_cb[TIMER_NUMOF];
+static void *_timer_ctx_arg[TIMER_NUMOF];
+static unsigned _timer_flag_periodic[TIMER_NUMOF];
+static unsigned _timer_flag_reset[TIMER_NUMOF];
+
+static inline uint64_t _timer_read_us(tim_t dev)
+{
+    (void)dev;
+    uint32_t time_h, tmp = TIMER->TIMERAWH;
+    uint32_t time_l;
+    do {
+        time_h = tmp;
+        time_l = TIMER->TIMERAWL;
+    } while ((tmp = TIMER->TIMERAWH) != time_h);
```

Comment:
The SDK does not do it that way because of the second core. But since we are only running on core 0, we could certainly just read TIMELR and then TIMEHR.